packages/linter/src/generators/convert-to-flat-config/generator.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  formatFiles,
+  getProjects,
+  NxJsonConfiguration,
+  ProjectConfiguration,
+  Tree,
+  updateJson,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+import { ConvertToFlatConfigGeneratorSchema } from './schema';
+import { findEslintFile } from '../utils/eslint-file';
+import { convertEslintJsonToFlatConfig } from './converters/json-converter';
+
+export async function convertToFlatConfigGenerator(
+  tree: Tree,
+  options: ConvertToFlatConfigGeneratorSchema
+) {
+  const eslintFile = findEslintFile(tree);
+  if (!eslintFile) {
+    throw new Error('Could not find root eslint file');
+  }
+  if (!eslintFile.endsWith('.json')) {
+    throw new Error(
+      'Only json eslint config files are supported for conversion'
+    );
+  }
+
+  // rename root eslint config to eslint.config.js
+  convertRootToFlatConfig(tree);
+  // rename and map files
+  const projects = getProjects(tree);
+  for (const [project, projectConfig] of projects) {
+    convertProjectToFlatConfig(tree, project, projectConfig);
+  }
+  // replace references in nx.json
+  updateNxJsonConfig(tree);
+  // install missing packages
+
+  if (!options.skipFormat) {
+    await formatFiles(tree);
+  }
+}
+
+export default convertToFlatConfigGenerator;
+
+function convertRootToFlatConfig(tree: Tree) {
```

Comment:
I feel like this function should accept the path to the root eslint file as an argument. We've already located it before calling the function.

```suggestion
function convertRootToFlatConfig(tree: Tree, rootEsLintFIlePath: string) {
```

Reply:
This function converts both `.eslintrc.base.json` and `.eslintrc.json`. The found `rootEsLintFIlePath` could be any one of those two. The second condition on line 56 is not needed, but we still need to check for base one.

But I guess it's cheaper to check the file name's suffix then check for file's existence